shm/Map.cpp

```diff
@@ -0,0 +1,42 @@
+#include "Map.hpp"
+#include <algorithm>
+#include <iostream>
+#include <random>
+#include <set>
```

Comment:
I think there should be new line between header include and standard library includes